springfox-bean-validators/src/main/java/springfox/bean/validators/plugins/Validators.java

```diff
@@ -119,12 +119,11 @@ public static boolean annotationMustBeApplied(ModelPropertyContext context, Clas
     }
 
     Set<Class<?>> validatedGroupsClasses = getGroupClasses(validatedGroups.stream().map(ResolvedType::getErasedType));
-    Set<Class<?>> constraintGroupClasses = getGroupClasses(Stream.of(groupsConstraint));
+    Set<Class<?>> constraintGroupClasses = Stream.of(groupsConstraint).collect(Collectors.toSet());
 
-    // Case #2: Validated groups contains Default.class
-    // Constrain groups contains Default.class
-    if (validatedGroupsClasses.contains(Default.class)) {
-      return constraintGroupClasses.isEmpty() || constraintGroupClasses.contains(Default.class);
+    // Case #2: constraintGroupClasses is empty
+    if (constraintGroupClasses.isEmpty()) {
+      constraintGroupClasses.add(Default.class);
     }
```

Comment:
### Second condition (or `Case #2`) - 124 line
If the constraint annotation (such as `@NotNull`) does not contain a group, this case is the same as using the default group (`Default.class`).